# Assigment 1

- Nama         : Ines Savitri
- Kode Peserta : PYTN-KS07-017

### Mendownload Data

In [2]:
import numpy as np
import pandas as pd

In [3]:
df_LC = pd.read_csv('london_crime_by_lsoa.csv')

df_LC.head()

lsoa_code     borough               major_category  \
0  E01001116     Croydon                     Burglary   
1  E01001646   Greenwich  Violence Against the Person   
2  E01000677     Bromley  Violence Against the Person   
3  E01003774   Redbridge                     Burglary   
4  E01004563  Wandsworth                      Robbery   

                minor_category  value  year  month  
0  Burglary in Other Buildings      0  2016     11  
1               Other violence      0  2016     11  
2               Other violence      0  2015      5  
3  Burglary in Other Buildings      0  2016      3  
4            Personal Property      0  2008      6

In [4]:
# Cleansing Data dengan Standar Missing Values

df_LC.isnull().sum

<bound method NDFrame._add_numeric_operations.<locals>.sum of          lsoa_code  borough  major_category  minor_category  value   year  \
0            False    False           False           False  False  False   
1            False    False           False           False  False  False   
2            False    False           False           False  False  False   
3            False    False           False           False  False  False   
4            False    False           False           False  False  False   
...            ...      ...             ...             ...    ...    ...   
1048570      False    False           False           False  False  False   
1048571      False    False           False           False  False  False   
1048572      False    False           False           False  False  False   
1048573      False    False           False           False  False  False   
1048574      False    False           False           False  False  False   

         mont

In [5]:
df_LC.isnull().values.any()

False

###### Berdasarkan keterangan di atas, data yang akan dipakai sudah bersih, tidak ada data yang hilang atau rusak, sehingga data dapat diolah.

In [6]:
df_LC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 7 columns):
 #   Column          Non-Null Count    Dtype 
---  ------          --------------    ----- 
 0   lsoa_code       1048575 non-null  object
 1   borough         1048575 non-null  object
 2   major_category  1048575 non-null  object
 3   minor_category  1048575 non-null  object
 4   value           1048575 non-null  int64 
 5   year            1048575 non-null  int64 
 6   month           1048575 non-null  int64 
dtypes: int64(3), object(4)
memory usage: 56.0+ MB


##### Berdasarkan keterangan di atas, dapat diketahui bahwa data berjumlah 7 kolom dan 1048575 baris, dengan tipe data object dan integer.

### Dataset ini berisi :

- lsoa_code      : mewakili kebijakan area
- borough        : wilayah london
- major_category : kategori kejahatan berat
- minor_category : kategori kejahatan ringan
- value          : jumlah kejahatan dalam wilayah tertentu berdasarkan bulan
- year           : tahun ringkasan statistik
- month          : bulan ringkasan statistik

In [7]:
# mendeskripsikan dataframe value
df_LC.describe().value

count    1.048575e+06
mean     4.753327e-01
std      1.678109e+00
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      1.000000e+00
max      2.410000e+02
Name: value, dtype: float64

In [8]:
# melihat 5 data teratas
df_LC.head(5)

lsoa_code     borough               major_category  \
0  E01001116     Croydon                     Burglary   
1  E01001646   Greenwich  Violence Against the Person   
2  E01000677     Bromley  Violence Against the Person   
3  E01003774   Redbridge                     Burglary   
4  E01004563  Wandsworth                      Robbery   

                minor_category  value  year  month  
0  Burglary in Other Buildings      0  2016     11  
1               Other violence      0  2016     11  
2               Other violence      0  2015      5  
3  Burglary in Other Buildings      0  2016      3  
4            Personal Property      0  2008      6

In [9]:
# melihat 5 data terbawah
df_LC.tail(5)

lsoa_code               borough             major_category  \
1048570  E01001649             Greenwich         Theft and Handling   
1048571  E01000464                Bexley            Criminal Damage   
1048572  E01000064  Barking and Dagenham                      Drugs   
1048573  E01004615            Wandsworth            Criminal Damage   
1048574  E01001752               Hackney  Other Notifiable Offences   

                      minor_category  value  year  month  
1048570                  Other Theft      3  2010     11  
1048571  Criminal Damage To Dwelling      0  2012      3  
1048572          Possession Of Drugs      1  2010      2  
1048573        Other Criminal Damage      1  2016      7  
1048574             Other Notifiable      0  2008     12

In [10]:
# untuk mendapatkan list header kolom
df_LC.columns.values

array(['lsoa_code', 'borough', 'major_category', 'minor_category',
       'value', 'year', 'month'], dtype=object)

In [11]:
# untuk mendapatkan daftar indeks
df_LC.index.values

array([      0,       1,       2, ..., 1048572, 1048573, 1048574],
      dtype=int64)

In [12]:
# melihat dimensi dataframe
df_LC.shape

(1048575, 7)

In [13]:
# menghapus beberapa kolom yang tidak perlu
df_LC.drop(['lsoa_code', 'month'], axis=1, inplace=True)
df_LC.head(2)

borough               major_category               minor_category  value  \
0    Croydon                     Burglary  Burglary in Other Buildings      0   
1  Greenwich  Violence Against the Person               Other violence      0   

   year  
0  2016  
1  2016

In [14]:
df_LC.drop(['Summary'], axis=1, inplace=True)
df_LC

KeyError: "['Summary'] not found in axis"

In [15]:
df_LC[['borough','year', 'value']]

borough  year  value
0                     Croydon  2016      0
1                   Greenwich  2016      0
2                     Bromley  2015      0
3                   Redbridge  2016      0
4                  Wandsworth  2008      0
...                       ...   ...    ...
1048570             Greenwich  2010      3
1048571                Bexley  2012      0
1048572  Barking and Dagenham  2010      1
1048573            Wandsworth  2016      1
1048574               Hackney  2008      0

[1048575 rows x 3 columns]

In [16]:
# Filtering based on a criteria
condition = df_LC['borough'] == 'Greenwich'
print(condition)

0          False
1           True
2          False
3          False
4          False
           ...  
1048570     True
1048571    False
1048572    False
1048573    False
1048574    False
Name: borough, Length: 1048575, dtype: bool


In [17]:
df_LC[condition]

borough               major_category  \
1        Greenwich  Violence Against the Person   
75       Greenwich                     Burglary   
100      Greenwich                        Drugs   
221      Greenwich              Criminal Damage   
248      Greenwich           Theft and Handling   
...            ...                          ...   
1048385  Greenwich              Criminal Damage   
1048420  Greenwich           Theft and Handling   
1048479  Greenwich                        Drugs   
1048512  Greenwich           Theft and Handling   
1048570  Greenwich           Theft and Handling   

                                 minor_category  value  year  
1                                Other violence      0  2016  
75                  Burglary in Other Buildings      0  2016  
100                         Possession Of Drugs      0  2012  
221            Criminal Damage To Motor Vehicle      0  2008  
248      Motor Vehicle Interference & Tampering      0  2009  
...                                         ...    ...   ...  
1048385       Criminal Damage To Other Building      0  2009  
1048420             Theft/Taking of Pedal Cycle      0  2011  
1048479                        Drug Trafficking      0  2014  
1048512                        Theft From Shops      1  2014  
1048570                             Other Theft      3  2010  

[32893 rows x 5 columns]

In [18]:
# meninjau perubahan yang sudah dibuat pada kerangka data
print('data dimensions:', df_LC.shape)
print(df_LC.columns)
df_LC.head(10)

data dimensions: (1048575, 5)
Index(['borough', 'major_category', 'minor_category', 'value', 'year'], dtype='object')


borough               major_category                     minor_category  \
0     Croydon                     Burglary        Burglary in Other Buildings   
1   Greenwich  Violence Against the Person                     Other violence   
2     Bromley  Violence Against the Person                     Other violence   
3   Redbridge                     Burglary        Burglary in Other Buildings   
4  Wandsworth                      Robbery                  Personal Property   
5      Ealing           Theft and Handling                        Other Theft   
6      Ealing  Violence Against the Person                   Offensive Weapon   
7    Hounslow                      Robbery                  Personal Property   
8      Newham              Criminal Damage  Criminal Damage To Other Building   
9      Sutton           Theft and Handling        Theft/Taking of Pedal Cycle   

   value  year  
0      0  2016  
1      0  2016  
2      0  2015  
3      0  2016  
4      0  2008  
5      0  2012  
6      0  2010  
7      0  2013  
8      0  2013  
9      1  2016

### Visualisasi menggunakan Matplotlib

In [19]:
%matplotlib inline

import matplotlib as mpl
import matplotlib.pyplot as plt

In [20]:
print('Matplotlib version: ', mpl.__version__)

Matplotlib version:  3.5.1


In [21]:
print(plt.style.available)
mpl.style.use(['ggplot'])

['Solarize_Light2', '_classic_test_patch', '_mpl-gallery', '_mpl-gallery-nogrid', 'bmh', 'classic', 'dark_background', 'fast', 'fivethirtyeight', 'ggplot', 'grayscale', 'seaborn', 'seaborn-bright', 'seaborn-colorblind', 'seaborn-dark', 'seaborn-dark-palette', 'seaborn-darkgrid', 'seaborn-deep', 'seaborn-muted', 'seaborn-notebook', 'seaborn-paper', 'seaborn-pastel', 'seaborn-poster', 'seaborn-talk', 'seaborn-ticks', 'seaborn-white', 'seaborn-whitegrid', 'tableau-colorblind10']


In [22]:
# Line plot

Greenwich = df_LC.loc['Greenwich']
Greenwich.head()

KeyError: 'Greenwich'